src/test/java/ClientTest.java

```diff
@@ -35,70 +35,73 @@ public void setUp() {
             ContentType contentType = ContentType.fromString(documentMimeType);
             String consentId = UUID.randomUUID().toString();
 
-            JSONObject document = this.client.postDocuments(contentType, consentId);
-            ClientTest.assertDocument(document);
+            JSONObject document = this.client.createDocument(content, contentType, consentId, new HashedMap());
+            Assert.assertTrue(document.has("consentId"));
+            Assert.assertTrue(document.has("contentType"));
+            Assert.assertTrue(document.has("documentId"));
         }
     }
 
     @Test
-    public void testPostPredictions() throws IOException, URISyntaxException {
+    public void testCreatePrediction() throws IOException, URISyntaxException {
         String[] modelNames = this.toArray(this.config.getProperty("model.names"));
         String[] documentPaths = this.toArray(this.config.getProperty("document.paths"));
         String[] documentMimeTypes = this.toArray(this.config.getProperty("document.mime.types"));
-        Assert.assertEquals(modelNames.length, documentPaths.length);
-        Assert.assertEquals(modelNames.length, documentMimeTypes.length);
 
         for (int i = 0; i < documentPaths.length; ++i) {
             String modelName = modelNames[i];
             String documentPath = this.getResourcePath(documentPaths[i]);
             String documentMimeType = documentMimeTypes[i];
+
+            byte[] content = UUID.randomUUID().toString().getBytes();
             ContentType contentType = ContentType.fromString(documentMimeType);
             String consentId = UUID.randomUUID().toString();
 
-            JSONObject document = this.client.postDocuments(contentType, consentId);
-            URI uploadUri = new URI(document.getString("uploadUrl"));
+            JSONObject document = this.client.createDocument(content, contentType, consentId, new HashedMap());
             String documentId = document.getString("documentId");
 
-            this.client.putDocument(documentPath, contentType, uploadUri);
             JSONObject prediction = this.client.postPredictions(documentId, modelName);
 
+            System.out.println("document: " + document);
+            System.out.println("prediction" + prediction);
             JSONArray fields = prediction.getJSONArray("predictions");
             Assert.assertNotNull(fields);
+
             StreamSupport.stream(fields.spliterator(), false)
                     .map(o -> (JSONObject)o)
-                    .forEach(ClientTest::assertField);
+                    .forEach(field -> {
+                        Assert.assertNotNull(field.get("label"));
+                        Assert.assertNotNull(field.get("value"));
+                        float confidence = field.getFloat("confidence");
+                        Assert.assertTrue(confidence >= 0.0);
+                        Assert.assertTrue(confidence <= 1.0);
+                    });
```

Comment:
Is this the correct indentation level? (looks excessive)

Reply:
It's correct.